vmas/simulator/environment/environment.py

```diff
@@ -770,6 +773,51 @@ def render(
         # render to display or array
         return self.viewer.render(return_rgb_array=mode == "rgb_array")
 
+    def plot_boundary(self):
+
+        # include boundaries in the rendering
+        from vmas.simulator.rendering import Line
+        from vmas.simulator.utils import Color
+
+        # check if the environment is dimension-limited
+        if self.world.x_semidim is not None or self.world.y_semidim is not None:
+            # set a big value for the cases where the environment is dimension-limited only in one coordinate
+            infinite_value = 100
+
+            x_semi = self.world.x_semidim if self.world.x_semidim else infinite_value
+            y_semi = self.world.y_semidim if self.world.y_semidim else infinite_value
+
+            # set the color for the boundary line
+            color = Color.GRAY.value
+
+            # Define boundary points based on whether world semidims are provided
+            if (self.world.x_semidim and self.world.y_semidim) or self.world.x_semidim:
```

Comment:
```suggestion
            if (self.world.x_semidim and self.world.y_semidim) or self.world.y_semidim is not None:
```

- i think it was y not x
- always use `is not None` as it might trigger or not when it is 0 or not (imagine the nightmare)

Reply:
Ok, should we add "is not None" also in the other parts?

Reply:
Apologies for the caos :(

I wanted to ask if I should include 'is not None' in other if-statements like:

- `x_semi = self.world.x_semidim if self.world.x_semidim else infinite_value` --> `x_semi = self.world.x_semidim if self.world.x_semidim is not None else infinite_value` (and similarly for `y_semi`)
- `1 if (self.world.x_semidim and self.world.y_semidim) else 2` --> `1 if (self.world.x_semidim is not None and self.world.y_semidim is not None) else 2`

I've tested all the scenarios with and without 'is not None,' and they worked fine.

But just to be on the safe side, as you suggested, I'll add 'is not None' wherever possible. Does that sound good?